python/kfserving/kfserving/api/kf_serving_client.py

```diff
@@ -102,6 +105,41 @@ def get(self, name, namespace=None):
             raise RuntimeError(
                 "Exception when calling CustomObjectsApi->get_namespaced_custom_object: %s\n" % e)
 
+    def watch(self, name=None, namespace=None, timeout_seconds=900):
+        """Watch the created KFService in the specified namespace"""
+
+        if namespace is None:
+            namespace = utils.get_default_target_namespace()
+
+        tbl = TableLogger(
+            columns='NAME,READY,DEFAULT TRAFFIC,CANARY TRAFFIC,URL',
+            colwidth={'NAME': 20, 'READY':7, 'DEFAULT TRAFFIC':15, 'CANARY TRAFFIC':14, 'URL': 45},
+            file=sys.stdout)
+
+        stream = k8s_watch.Watch().stream(
+            client.CustomObjectsApi().list_namespaced_custom_object,
```

Comment:
instead of listing all custom objects and then filter you can use `label_selector` to direct;y get the specified `KFService`

Reply:
@yuzisun This is not list all custom objects, but just KFService, seems the `k8s.watch.Watch().stream` the first arg is function name, and following is the function args, you can see the 2nd, 3rd args .. Thanks.

Reply:
ye, I mean you can pass `label_selector="serving.kubeflow.org/kfservice=flowers-sample"` as the function arg which is the on the argument list of `list_namespaced_custom_object`

Reply:
@yuzisun  Oh.. understood. But I do not think we can use the `label_selector` unless we defined the metadata.labels when creating the kfservice or by default force adding name to metadata as key/value while creating that. obvious not, and we also cannot ensure user add the labels.
I guess you are worried about performance if get all? in fact I think that's OK since just `watch` just mounting events.

Reply:
might be a good idea to add a label - would also help us identify other objects that are related

Reply:
@jinchihe check here we already have a label for this. 
https://github.com/kubeflow/kfserving/blob/master/pkg/controller/kfservice/resources/knative/configuration.go#L104

Reply:
@yuzisun From my understanding, we added the label to `pod` of the kfservice, not for `kfservice`. The api claims `kfservice` level. Double checked the created `kfservice`, no labels while pod have. please correct me if anymistake. Thanks.

Reply:
My bad, thought it is on the KFService also.